lib/reek/configuration/configuration_converter.rb

```diff
@@ -101,10 +103,25 @@ def strings_to_regexes_for_directories
                   to_regex item
                 end
               end
+            rescue NoMethodError
+              warn_about_missing_configuration(detector) if configuration.nil?
```

Comment:
I think it's better to detect that `configuration` is `nil` at the beginning of this block and handle that specific case there.